src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -126,6 +115,30 @@ export default class DefaultMessagingSession implements MessagingSession {
       } else {
         this.reconnectController.startedConnectionAttempt(true);
       }
+
+      // reconnect needs to re-resolve endpoint url, which will also refresh credentials on client if they are expired.
+      let endpointUrl = !reconnecting ? this.configuration.endpointUrl : undefined;
+      if (endpointUrl === undefined) {
+        try {
+          const endpoint = await this.configuration.chimeClient
+            .getMessagingSessionEndpoint()
+            .promise();
+          endpointUrl = endpoint.Endpoint.Url;
+        } catch (e) {
+          const closeEvent = new CloseEvent('close', {
```

Comment:
Could you print the error here before creating the CloseEvent?
```
this.logger.error(${error});
```

Reply:
This is already logged when handled in the closeEventHandler so don't think duplicate logging is needed here: `  this.logger.info(`WebSocket close: ${event.code} ${event.reason}`);`

Reply:
Yeah, but the reason is logged, not sure if `error` can be something different hence suggested.

Reply:
Also, good solution I think.